### PR TITLE
Using the new bakery protocol version and upgrade to httprequest.

### PR DESCRIPTION
**Pull request by fabricematrat**

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/1135/
Test PASSed.

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/1136/
Test FAILed.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/1137/
Test PASSed.

**Comment by jrwren** [comment]
:+1:

File: internal/v4/api_test.go [hunk 1]
```diff
@@ -898,6 +898,20 @@ func (s *APISuite) TestExtraInfoPutUnauthorized(c *gc.C) {
 		ExpectStatus: http.StatusProxyAuthRequired,
 		ExpectBody:   dischargeRequiredBody,
 	})
+	httptesting.AssertJSONCall(c, httptesting.JSONCallParams{
```
Comment (rogpeppe) [inline internal/v4/api_test.go:901]: This should probably be asserting that the response header contains the appropriate WWW-Authorize header. Mind you, it's not that crucial as the tests failed anyway when it didn't.

**Comment by rogpeppe** [comment]
LGTM with one minor thought. Thanks!

**Comment by jrwren** [comment]
:+1:

**Comment by jrwren** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/charmstore-merge

**Pull request merged**